util/metrics/src/service_metrics.rs

```diff
@@ -100,17 +119,25 @@ impl ServiceMetrics {
         )
     }
 
+    /// Count number of errors by method
     pub fn resp(&self, ctx: &RpcContext, success: bool) {
-        // The reason for counting everything here, instead of doing the
-        // if outside of the increment is that we could also compare
-        // number of responses to number of requests
         if let Some(name) = path_from_ctx(ctx) {
             self.num_error
                 .with_label_values(&[name.as_str()])
                 .inc_by(if success { 0 } else { 1 });
         }
     }
 
+    /// Count number of response codes by method
+    pub fn status_code(&self, ctx: &RpcContext, response_code: RpcStatusCode) {
```

Comment:
Should this method be renamed to something like `count_number_of_response_codes_per_method`? The name `status_code` is somewhat vague and doesn't describe what's going on (unless I'm misunderstanding and this is like some trait or gRPC construct).

Reply:
That might be too verbose, when modifying libraries like this I try to keep the semantics close to what the other methods have. I could do "count_status_codes"